components/ILIAS/Contact/classes/class.MailSearchObjectMembershipsTable.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data;
+use ILIAS\UI;
+use Psr\Http\Message\ServerRequestInterface;
+
+class MailSearchObjectMembershipsTable implements UI\Component\Table\DataRetrieval
+{
+    /** @var array<string, string> */
+    private readonly array $mode;
+    private readonly Data\Factory $data_factory;
+    private ServerRequestInterface|\Psr\Http\Message\RequestInterface $request;
+    /** @var array<int, array<string, string>>|null */
+    private ?array $records = null;
+    private bool $buddysystem_enabled;
+    private bool $mailing_allowed = false;
+
+    private function isMailingAllowed(): bool
+    {
+        return $this->mailing_allowed;
+    }
+
+    public function setMailingAllowed(bool $mailing_allowed): void
+    {
+        $this->mailing_allowed = $mailing_allowed;
+    }
+
+    private function isBuddysystemEnabled(): bool
+    {
+        return $this->buddysystem_enabled;
+    }
+
+    public function __construct(
+        private readonly array $obj_ids,
```

Comment:
We should describe the shape of `$obj_ids`. I assume this is a list of `int` values.